tests/cargo-kani/no_std/src/main.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! This test makes sure Kani handles assert in no_std environment which was
+//! previous failing (see https://github.com/model-checking/kani/issues/2187)
+
+#![no_std]
+
+#[cfg(feature = "std")]
+extern crate std;
+
+#[kani::proof]
+fn foo() {
+    let x: i32 = kani::any();
+    let y = 0;
+    std::debug_assert!(x + y == x, "Message");
```

Comment:
Should we also have a test where the assertion fails?

Reply:
I don't think this is necessary, since we're mainly checking that compilation goes through. There are many other tests that check for failing assertions.